clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -892,7 +893,11 @@ protected synchronized String memberId() {
 
     private synchronized void resetGeneration() {
         this.generation = Generation.NO_GENERATION;
-        resetStateAndRejoin();
+        rejoinNeeded = true;
```

Comment:
Leaving comment here for lack of an alternative. Is it worth checking for the case of an empty assignment? Would the client have been able to recover from this problem by leaving the group and rejoining? I am not sure how much value trying to detect broker versions which are affected by this bug. At least perhaps we can provide a better error message than the BufferUnderflow.

Reply:
Yeah, I had the same thought. I'm satisfied with a better error message.